src/rust/engine/process_execution/src/local.rs

```diff
@@ -710,11 +712,15 @@ pub async fn prepare_workdir(
   // non-determinism when paths overlap: see the method doc.
   let store2 = store.clone();
   let workdir_path_2 = workdir_path.clone();
+  let mut mutable_paths = req.output_files.clone();
+  mutable_paths.extend(req.output_directories.clone());
```

Comment:
Can you open a ticket for followup to replace the `Process.immutable_inputs` argument with the include/exclude list arguments?